Junk4Dollars-appTests/ViewControllerTests.swift

```diff
@@ -0,0 +1,51 @@
+import XCTest
+import UIKit
+import Junk4Dollars_app
+
+
+class ViewControllerTests: XCTestCase {
+    func testLabelDefaultValue() {
+        let controller = ViewController()
+        // Need to manually instantiate a new label here using UIKit
+        controller.label = UILabel()
+        controller.label.text = "foo"
+        
+        controller.viewDidLoad()
+        
+        XCTAssertEqual("", controller.label.text)
+    }
+    
+    func testSayHelloDefault() {
+        let controller = ViewController()
+        controller.label = UILabel()
+        controller.textInput = UITextField()
+        
+        controller.tapSayHello()
+        
+        XCTAssertEqual("Hello!", controller.label.text)
+    }
+    
+    func testSayHelloWithName() {
+        let controller = ViewController()
+        controller.label = UILabel()
+        controller.textInput = UITextField()
+        controller.textInput.text = "John"
+        
+        controller.tapSayHello()
+        
+        XCTAssertEqual("Hello, John!", controller.label.text)
+    }
+    
+    func testResetButton() {
```

Comment:
Same comment as with the UI test, test names should explain the scenario + expected result, so you don't have to dig deep into the test to understand what it's doing.